typescript/sdk/src/core/app.ts

```diff
@@ -36,6 +37,16 @@ export class AbacusCore<Chain extends ChainName = ChainName> extends AbacusApp<
     return new AbacusCore(environments[name], multiProvider);
   }
 
+  extendWithConnectionManagers<T>(
+    config: ChainMap<Chain, T>,
+  ): ChainMap<Chain, T & { abacusConnectionManager: string }> {
+    return objMap(config, (chain, config) => ({
+      ...config,
+      abacusConnectionManager:
+        this.getContracts(chain).abacusConnectionManager.address,
+    }));
+  }
+
```

Comment:
not sure I agree this should be on the `app` abstraction

Reply:
Where should this be instead? Do you think people should pluck it out from the addresses config for core themselves? Shouldn't AbacusCore be the interface by which you interact with the abacus core contracts?

Reply:
So this takes a config map, adds properties to it, and then returns it without any side effects? I think I agree with York that this feels more like a util function which could take `CoreContractSchema<Chain, Local>` as a 2nd param. I'm probably missing context though

Reply:
agree this is just a config utils
maybe on the deployer or elsewhere?
it doesn't seem specific to core, any `RouterApp` could provide this right?

Reply:
It's specific to core because the addresses come from core? How would the utils function get the addresses otherwise?

Reply:
@nambrot They would call `this.getContracts` first, but maybe that's not worth the hassle and keeping it here is better